crates/next-core/src/next_server/context.rs

```diff
@@ -107,6 +108,26 @@ pub async fn get_server_resolve_options_context(
                 ..resolve_options_context
             }
         }
+        ServerContextType::Middleware => {
+            let resolve_options_context = ResolveOptionsContext {
+                enable_node_modules: true,
+                enable_node_externals: true,
+                enable_node_native_modules: true,
+                module: true,
+                custom_conditions: vec!["development".to_string()],
+                import_map: Some(next_server_import_map),
```

Comment:
This import map probably defines some externals which we don't want here.

There must be no externals in an edge compilation.